### PR TITLE
Add code quality chapter

### LINKED ISSUE #882: Add Code Quality chapter
## Detailed description
 * This issue is a place to discuss matters relating to the writing of the chapter on code quality.
 * This chapter is being written in #898
 * This chapter is being developed on the _chapter_topic_ branch.

### Table of contents
1. Tools for checking software quality (linters)
2. Online services providing software quality checks
3. Coding style
4. Automatic formatters

### Resources
Currently we have this chapter i the guide: https://guide.esciencecenter.nl/best_practices/code_quality.html

## Current status
Currently we have this chapter in the guide: https://guide.esciencecenter.nl/best_practices/code_quality.html. I would like to add the contents to the Turing Way if this seems suitable.

 - [ ] Write chapter outline
 - [ ] Add material to the chapter
 - [ ] Combine materials into a readable chapter
 - [ ] Proofread
 - [ ] Request reviews
 - [ ] Address reviews
 - [ ] Merge to master branch.

## Updates
To avoid that others have to read through the full thread of comments, update the initial issue with important updates (e.g. decisions taken) regularly so that all the important information is available at one glance.

### PR DESCRIPTION
**Pull request by bouweandela**
### Summary

Fixes #882

### List of changes proposed in this PR (pull-request)

* *Add a code quality chapter.*

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

File: book/content/reviewing/03/typical_workflows.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Before requesting a review, be sure you've met all the obvious quality benchmark
 - you have created [documentation](#Documentation) to the required standards of the project,
```
Comment (mkuzak) [inline book/content/reviewing/03/typical_workflows.md]: "documentation of the required standards" or "documentation about the required standards"

File: book/content/version_control/version_control.md [hunk 2]
```diff
@@ -723,7 +723,7 @@ This also works for using commit messages to close issues too.
 
 ## What to learn next
 
-Look into best practice for writing good quality code (for example, good naming conventions, informative comments, modular code structure).
+Look into [best practice for writing good quality code](../code_quality/code_quality) (for example, good naming conventions, informative comments, modular code structure).
```
Comment (mkuzak) [inline book/content/version_control/version_control.md]:
```suggestion
Look into [best practice for writing good quality code](../code_quality/code_quality) (eg. good naming conventions, informative comments, modular code structure).
```

**Comment by mkuzak** [comment]
@malvikasharan this can be merged after few small suggestions have been addressed

File: book/content/version_control/version_control.md (hunk 2 above)
Comment (malvikasharan) [reply book/content/version_control/version_control.md]: The style guide and CI don't like Latin abbreviations :(

File: book/website/_data/toc.yml [hunk 3]
```diff
@@ -117,6 +117,7 @@
     url: reproducible_environments/08/resources
 - title: Code quality
   url: code_quality/code_quality
+    url: /reproducible_environments/08/resources
```
Comment (malvikasharan) [inline book/website/_data/toc.yml]:
```suggestion
```

**Comment by malvikasharan** [comment]
@bouweandela and @mkuzak This is ready for merge. 🎊🎉

**Pull request merged**